pkg/controller/servicebindingrequest/servicebinder_test.go

```diff
@@ -573,3 +572,48 @@ func TestServiceBinder_Bind(t *testing.T) {
 		},
 	}))
 }
+
+func TestEnsureDefaults(t *testing.T) {
+	t.Run("label selector with non nil", func(t *testing.T) {
+		applicationSelector := &v1alpha1.ApplicationSelector{}
+		ensureDefaults(applicationSelector)
+		require.NotNil(t, applicationSelector.LabelSelector)
+	})
+
+	t.Run("empty label selector", func(t *testing.T) {
+		applicationSelector := &v1alpha1.ApplicationSelector{}
+		require.Nil(t, applicationSelector.LabelSelector)
+		ensureDefaults(applicationSelector)
+		require.NotNil(t, applicationSelector.LabelSelector)
+	})
+
+	t.Run("default pod spec path", func(t *testing.T) {
+		applicationSelector := &v1alpha1.ApplicationSelector{}
+		ensureDefaults(applicationSelector)
+		containersPath := getContainersPath(applicationSelector)
+		expectedContainersPath := []string{"spec", "template", "spec", "containers"}
+		require.Equal(t, expectedContainersPath, containersPath)
+	})
+
+	t.Run("container path with value", func(t *testing.T) {
+		applicationSelector := &v1alpha1.ApplicationSelector{}
+		applicationSelector.BindingPath = &v1alpha1.BindingPath{
+			ContainersPath: "spec.some.path",
+		}
+		ensureDefaults(applicationSelector)
+		containersPath := getContainersPath(applicationSelector)
+		expectedContainersPath := []string{"spec", "some", "path"}
+		require.Equal(t, expectedContainersPath, containersPath)
+	})
+	t.Run("container path with secret value", func(t *testing.T) {
+		applicationSelector := &v1alpha1.ApplicationSelector{}
+		applicationSelector.BindingPath = &v1alpha1.BindingPath{
+			SecretPath: "spec.some.path",
+		}
+		ensureDefaults(applicationSelector)
+		containersPath := getContainersPath(applicationSelector)
+		expectedContainersPath := []string{"spec", "template", "spec", "containers"}
+		require.Equal(t, expectedContainersPath, containersPath)
```

Comment:
Does `application.BindingPath.SecretPath` still contain the value `spec.some.path`?

Reply:
After my recent commit (9e875534459af740b5ac4a7b3c8e4cc48bd60069), it should be working. 

I have added a test here: 45f1522751783eeaa2848b5d94a849af874388c3